src/components/FilterBar.test.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { Suspense } from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { render, screen, waitFor } from '@testing-library/react';
+import UserEvent from '@testing-library/user-event';
+import FilterBar from './FilterBar';
+import { I18nextProvider } from 'react-i18next';
+import i18n from '../i18n';
+import { act } from 'react-dom/test-utils';
+
+afterAll(() => {
+  jest.clearAllMocks();
+});
+
+function renderComponent(filterParams = '') {
+  return render(
+    <Suspense fallback="loading">
+      <I18nextProvider i18n={i18n}>
+        <MemoryRouter initialEntries={[filterParams]}>
+          <FilterBar openFilter={() => {}} />
+        </MemoryRouter>
+      </I18nextProvider>
+    </Suspense>
+  );
+}
+
+test('renders filters & sort button', async () => {
+  renderComponent();
+  await waitFor(() => expect(screen.getByText('Filters')).toBeInTheDocument());
+  expect(screen.getByText('Sort')).toBeInTheDocument();
+});
+
+test('sort options dropdown', async () => {
+  renderComponent();
+  const sortBtn = await waitFor(() =>
+    screen.getByRole('button', { name: 'Sort' })
+  );
+  await UserEvent.click(sortBtn);
+
+  expect(
+    screen.getByRole('menuitem', { name: 'Amount (Low to High)' })
+  ).toBeInTheDocument();
+  expect(
+    screen.getByRole('menuitem', { name: 'Amount (High to Low)' })
+  ).toBeInTheDocument();
+  expect(
+    screen.getByRole('menuitem', { name: 'Deadline (Earliest to Latest)' })
+  ).toBeInTheDocument();
+  expect(
+    screen.getByRole('menuitem', { name: 'Deadline (Latest to Earliest)' })
+  ).toBeInTheDocument();
+});
+
+test('renders filters with # of filters applied', async () => {
+  renderComponent('?grades[]=8,9&minAmount=400&majors[]=test');
+  await waitFor(() =>
+    expect(screen.getByText('Filters (4)')).toBeInTheDocument()
+  );
+});
+
+test('translated filters & sort button - Spanish', async () => {
+  renderComponent();
+  await act(() => i18n.changeLanguage('es').then());
+
+  expect(screen.getByText('Filtros')).toBeInTheDocument();
```

Comment:
There might be a race condition in these kinds of these kinds of then that's causing the failure. You might be doing a `getBy` when the translation changes haven't yet been rerendered. Try doing something like this instead:
```suggestion
  await act(() => i18n.changeLanguage('es').then(() => screen.findByText('Filtros')));
```